src/ShopingCenter.java

```diff
@@ -0,0 +1,150 @@
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ShopingCenter {
+
+// *********** Static fields **********
+
+    private static int nextId = 0;
+
+// *********** End of Static fields ***********
+
+// *********** Fields ************
+
+    private String name;
+    private Address address;
+    private List<Shop> shopList;
+    private OpeningHours timeOpen;
+    private int id;
+
+    public ShopingCenter(String name, String address, int openTime, int closeTime) {
```

Comment:
we could use `this(name, new Address(address), new OpeningHours(openTime, closeTime));` here instead of full implementation

Reply:
Good idea, forgot myself for a moment!